From: Russ Allbery <eagle@eyrie.org>
Date: Sun, 13 Nov 2016 10:13:21 -0800
Subject: Fix OpenSSL library probe

OpenSSL 1.1 doesn't have SSL_library_init as a regular function.
Use a different function to  probe for the OpenSSL library.
---
 m4/openssl.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/m4/openssl.m4 b/m4/openssl.m4
index e5cd371..83f27be 100644
--- a/m4/openssl.m4
+++ b/m4/openssl.m4
@@ -70,10 +70,10 @@ AC_DEFUN([_RRA_LIB_OPENSSL_INTERNAL],
         [AC_MSG_ERROR([cannot find usable OpenSSL crypto library])])],
     [$rra_openssl_extra])
  AS_IF([test x"$rra_reduced_depends" = xtrue],
-    [AC_CHECK_LIB([ssl], [SSL_library_init], [OPENSSL_LIBS=-lssl],
+    [AC_CHECK_LIB([ssl], [SSL_accept], [OPENSSL_LIBS=-lssl],
         [AS_IF([test x"$1" = xtrue],
             [AC_MSG_ERROR([cannot find usable OpenSSL library])])])],
-    [AC_CHECK_LIB([ssl], [SSL_library_init],
+    [AC_CHECK_LIB([ssl], [SSL_accept],
         [OPENSSL_LIBS="-lssl $CRYPTO_LIBS"],
         [AS_IF([test x"$1" = xtrue],
             [AC_MSG_ERROR([cannot find usable OpenSSL library])])],
